pkg/asset/agent/image/agentimage.go

```diff
@@ -86,9 +86,18 @@ func (a *AgentImage) Generate(ctx context.Context, dependencies asset.Parents) e
 		return fmt.Errorf("AgentWorkflowType value not supported: %s", agentWorkflow.Workflow)
 	}
 
+	var err error
+	if a.tmpPath, err = agentArtifacts.PrepareArtifacts(ctx); err != nil {
+		return err
+	}
+	defer func(err error) {
+		if err != nil {
+			os.RemoveAll(a.tmpPath)
+		}
+	}(err)
```

Comment:
I'm not sure this would work, since it will capture the value of `err` when `defer` is defined (so it could be always `nil`). Capturing by closure should fix it

```suggestion
	}()
```